code/game/objects/structures/mirror.dm

```diff
@@ -61,6 +65,19 @@
 	if(I.damtype == STAMINA)
 		return
 	if(shattered)
+		if(istype(I, /obj/item/weapon/weldingtool))
+			var/obj/item/weapon/weldingtool/WT = I
+			if(WT.remove_fuel(0, user))
+				user << "<span class='notice'>You begin repairing the shaterred [src]...</span>"

```

Comment:
This would result in "You begin repairing the shattered the mirror" unless the name has an \improper before it or you change the message.